srcjs/_start.js

```diff
@@ -3,6 +3,17 @@
 
   var exports = window.Shiny = window.Shiny || {};
 
+  (function(history){
+    var pushState = history.pushState;
+    history.pushState = function(state) {
+        if (typeof history.onpushstate == "function") {
```

Comment:
This is too early. The original history.pushState hasn't been called yet, so the URL hasn't yet changed. That's the cause of your behind-by-one bug.